ecl/hql/hqlopt.cpp

```diff
@@ -3977,7 +3983,17 @@ IHqlExpression * CTreeOptimizer::doCreateTransformed(IHqlExpression * transforme
                     break;
                 }
             }
+            break;
+        }
+    case no_unordered:
+        //This is only added if not shared.  A future transform will optimize shared expressions where all uses are unordered.
+        if (!hasOrderedAttribute(child))
+        {
+            OwnedHqlExpr unorderedChild = appendOwnedOperand(child, getOrderedAttribute(false));
+            return replaceChild(transformed, unorderedChild);
         }
+        //MORE: Remove UNORDERED(SORT(ds,local|grouped))
```

Comment:
Putting Jira references in the code has not been normal practice (and I'm not sure I would want it to be)